### PR TITLE
Add possibility to push specific branch

### PR DESCRIPTION
**Pull request by Hugal31**
Add possibility to push a specific git branch
------------------------------------------------------
Use ```-b [branch]``` or ```--branch [branch]``` to specify the branch to push.

Git ftp will checkout to this branch, do his stuff and checkout again to the current working branch.

### CODE REVIEW

File: git-ftp [hunk 1]
```diff
@@ -845,6 +851,20 @@ set_remote_password() {
 	[ -z "$REMOTE_PASSWD" ] && REMOTE_PASSWD="$(get_config password)"
 }
 
+set_branch() {
+    [ -z $BRANCH ] && BRANCH="$(get_config branch)"
+    if [ ! -z $BRANCH ]; then
+	CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
```
Comment (mkllnk) [inline git-ftp]: This doesn't work if you have a detached HEAD, for example you are on origin/master. I solved it this way: https://github.com/git-ftp/git-ftp/blob/pull/git-ftp#L986

**Comment by mkllnk** [comment]
Thank you for the PR. I'm not sure how many people will need this, but it seems logical since you can push other branches with git as well: `git push origin my-local-branch:my-remote-branch`

And I see your use case with the prod branch. You still need a clean working directory though. In which case this new option saves you two checkout commands.

**Comment by Hugal31** [comment]
Thanks for your answer.

This modification is useful when I'm working on a project with a prod dev branch and a prod branch:
in the worst cases, this prevents me from accidentally pushing the dev branch on the prod server if I forget to checkout.

**Comment by geertbeskers** [comment]
@Hugal31 I like the possibility to push ftp with a specific branch. Is this already implemented in de current version? I can't find it.

**Comment by Hugal31** [comment]
@geertbeskers This isn't implemented in the git-ftp/git-ftp's current version. That's why I made this pull request

**Pull request merged**